### ISSUE TITLE
Changes correctly spelled words

### ISSUE THREAD

**Issue by hsponberg** [atom/autocomplete-plus#1018]
### Description

Correctly spelled words are being changed to other longer words that I don't want.

For example, "technology" is changed to "technologically"

This happens when I want to go to a newline, without adding a period or other punctation at the end of the line. This often happens for bulleted lists. I know I can change the Keymap for Confirming A Suggestion to only Tab, but I do want it to correct misspelled words on Enter

### Steps to Reproduce

1. Type the word "technology"
2. Hit enter

**Expected behavior:** the word "technology" is left unchanged

**Actual behavior:** the word changes to "technologically"

**Reproduces how often:** 100%

### Versions

autocomplete-plus: 2.41.0

**Comment by hsponberg** [comment]
Not just the enter case, this happens for using tab to confirm a suggestion as well. For example, if I want to type the word technology and then a tab, it changes to technologically.

**Comment by rsese** [comment]
Thanks for clarifying - for this situation where the setting you have to confirm a suggestion can conflict with what you want to do (i.e. actually tab or enter),  this sounds like what's described in https://github.com/atom/autocomplete-plus/issues/943.  We'll close in favor of that issue and you can subscribe there if you'd like (though let me know if misunderstood what you've described).

**Comment by hsponberg** [comment]
Hi @rsese, thanks for that consideration, but this issue is different, and that what I've reported is a bug and not a user preference for confirming a suggestion.

The reason is because while I am typing the word technology, up until I type the final 'y', the word is the top suggestion. For example, if I type "technolog" and then hit tab or enter, it places with the word "technology". However, if I completely type out the word, "technology" then it is removed from the suggestion list altogether. But it should still be the top suggestion - I've correct spelled that word, so it should be the obvious and first choice. 

Users expect that if they type a properly spelled word and then hit enter or tab that it will not be replaced with a different word.

**Comment by rsese** [comment]
Oh ok I think i see what you're saying - are these steps to reproduce accurate?

1. create a new file with this content:

```text
technology is changed to technologically
```
2. hit enter and type `technology`

At this point, you expect `technology` to be an autocomplete suggestion, actual behavior is `technologically` is the only suggestion?

**Comment by hsponberg** [comment]
Yes, exactly. And not just any suggestion but the first suggestion, so if I
hit (enter | tab | keystroke) the word I have correctly spelled isn't
changed. I think by default if I have correctly typed a word then it should
require me to take an additional step (like the down arrow) to select a
different suggestion.

You can also reproduce with common words such as dog (changed to doing) or
doing (changed to dotting) or dotting (changed to demonstrating)

**Comment by rsese** [comment]
Thanks for clarifying - I saw this old pull request (https://github.com/atom/autocomplete-plus/pull/547) that seems like it was supposed to make it so an exact match will be removed if it's the only suggestion, so I think I would expect "technology" to still be an option in our example above?  Something could have changed in the meantime of course.
